### PR TITLE
[Merged by Bors] - feat(MeasureTheory): integral_comp for LinearIsometryEquiv 

### PR DESCRIPTION
**Pull request by mcdoll**
Composition with a LinearIsometryEquiv on a finite dimensional real Hilbert space preserves integrals.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dupuisf** [comment]
Can you try generalizing this to `LinearIsometryEquivClass`?

**Comment by mcdoll** [comment]
I would love to but that requires restructuring the whole `LinearIsometry` file so that all the theorems are phrased for the classes and not the structure. Right now the class exists, but is basically useless. This PR was basically the reason for this experiment https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Bundled.20sets.2Ffunctions.20with.20even.20more.20classes and I would prefer that we have a discussion on whether classes or structures are the way to go forward before starting to rewrite existing parts of mathlib.

**Comment by sgouezel** [comment]
bors d+
Let's get this in, and revisit later if one can use classes of maps instead.

**Comment by mcdoll** [comment]
bors merge

File: Mathlib/MeasureTheory/Measure/Haar/InnerProductSpace.lean [hunk 1]
```diff
@@ -20,7 +20,7 @@ the canonical `volume` from the `MeasureSpace` instance.
 
 open FiniteDimensional MeasureTheory MeasureTheory.Measure Set
 
-variable {ι F : Type*}
+variable {ι E F A : Type*}
```
Comment (mcdoll) [inline Mathlib/MeasureTheory/Measure/Haar/InnerProductSpace.lean:23]:
```suggestion
variable {ι E F : Type*}
```

**Comment by mcdoll** [comment]
bors merge

**Pull request closed without merging**